WordPress/Classes/ViewRelated/Blog/SharingViewController.m

```diff
@@ -241,15 +241,46 @@ - (NSManagedObjectContext *)managedObjectContext
     return self.blog.managedObjectContext;
 }
 
+-(void)syncServices
+{
+    // Optimistically sync the sharing buttons.
+    [self syncSharingButtonsIfNeeded];
+    
+    // Refreshes the tableview.
+    [self refreshPublicizers];
+    
+    // Syncs servcies and connections.
+    [self syncPublicizeServices];
+    
+}
+- (void)setupReachability
+{
+    self.reachability = [Reachability reachabilityForInternetConnection];
+    [self.reachability startNotifier];
+}
+
+-(void)showConnectionError
+{
+    NSString *title = NSLocalizedString(@"No Connection", @"Title of error prompt when no internet connection is available.");
+    NSString *message = NSLocalizedString(@"The Internet connection appears to be offline.", @"Error message shown when a media upload fails because the user isn't connected to the internet.");
+    dispatch_async(dispatch_get_main_queue(), ^{
+        [WPError showAlertWithTitle:title message:message];
+    });
+}
+
 - (void)syncPublicizeServices
 {
     SharingService *sharingService = [[SharingService alloc] initWithManagedObjectContext:[self managedObjectContext]];
     __weak __typeof__(self) weakSelf = self;
     [sharingService syncPublicizeServicesForBlog:self.blog success:^{
         [weakSelf syncConnections];
     } failure:^(NSError *error) {
-        [SVProgressHUD showDismissibleErrorWithStatus:NSLocalizedString(@"Publicize service synchronization failed", @"Message to show when Publicize service synchronization failed")];
-        [weakSelf refreshPublicizers];
+        if (!weakSelf.reachability.isReachable) {
```

Comment:
This is a great step in the right direction! 😄 It works in my testing.

There is a comment on line 301 that reads as follows:
```
    // Sync sharing buttons if they have never been synced. Otherwise, the
    // management vc can worry about fetching the latest sharing buttons.
```

This got me to thinking - is there anything the user can actually do on this screen that requires the "internet offline" message? In my testing, I observed that any attempted changes on the _Manage_ screen trigger save attempts that already evince the following error:
<img width="504" alt="11001a" src="https://user-images.githubusercontent.com/221062/52677676-b7f12980-2ee3-11e9-9621-0385d3f30916.png">

Is that sufficiently friendly? If so, do we _need_ this additional reachability reminder?

Reply:
That's a great point for this PR and for the general discussion on when should we message the user. This PRs intention is to not show the publicized services failure message when the call fails due to lack of connectivity. How about we keep the reachability check but remove showConnectionError? Is this still a step in the right direction?

Reply:
> How about we keep the reachability check but remove showConnectionError?

To confirm, are you proposing that we remove the call on line 279?

Reply:
I was proposing we only show the publicized services error if it fails with available connectivity, but something doesn't feel right about this approach tbh. I wish we could just not show that message at all (not sure how it benefits the user.

Reply:
Right!

To be clear, the reachability check you added works, but in this case, I wondered if it was necessary this is really a jumping off point to ~5 separate screens that already appear to apply reachability checks. It is _absolutely_ a step in the right direction to remove the unfriendly HUD warning from `-[SharingViewController syncPublicizeServices]`.